packages/embark/src/lib/modules/tests/test.js

```diff
@@ -351,26 +371,27 @@ class Test {
     });
   }
 
-  static getWeb3Contract(contract, web3) {
-    const newContract = new EmbarkJS.Blockchain.Contract({
-      abi: contract.abiDefinition,
-      address: contract.deployedAddress,
-      from: contract.deploymentAccount || web3.eth.defaultAccount,
-      gas: constants.tests.gasLimit,
-      web3: web3
-    });
+  getEmbarkJSContract(contract, cb) {
+    const codeToRun = `
+      const newContract = new EmbarkJS.Blockchain.Contract({
+        abi: ${JSON.stringify(contract.abiDefinition)},
+        address: "${contract.deployedAddress || ""}" || undefined,
+        from: "${contract.deploymentAccount || ""}" || web3.eth.defaultAccount,
+        gas: "${constants.tests.gasLimit}",
+        web3: web3
+      });
 
-    newContract.filename = contract.filename;
-    if (newContract.options) {
-      newContract.options.from = contract.deploymentAccount || web3.eth.defaultAccount;
-      newContract.options.data = contract.code;
-      if (!newContract.options.data.startsWith('0x')) {
-        newContract.options.data = '0x' + newContract.options.data;
+      newContract.filename = "${contract.filename}";
+      if (newContract.options) {
+        newContract.options.from = "${contract.deploymentAccount || ""}" || web3.eth.defaultAccount;
+        newContract.options.data = "${contract.code}";
+        if (newContract.options.data && !newContract.options.data.startsWith('0x')) {
+          newContract.options.data = '0x' + newContract.options.data;
+        }
+        newContract.options.gas = "${constants.tests.gasLimit}";
       }
-      newContract.options.gas = constants.tests.gasLimit;
-    }
-
-    return newContract;
+      return newContract;`;
+    this.events.request("runcode:eval", codeToRun, cb, false, true); 
```

Comment:
Why do you need to eval the contracts in the VM? Why don't you just return the contract instance?

Reply:
If it's only for `web3`, I removed the need to pass web3 in my PR

Reply:
This is one of the main reasons for the PR. The contract needs to be pulled out of the VM because EmbarkJS in the VM is set up correctly and we are returning a new instance of an EmbarkJS contact.